File: src/main/java/eotree/data/EOByte.java
package eotree.data;

import eotree.EONode;

/**
 * EBNF representation:
 * <code>
 * /[\dA-F][\dA-F]/
 * </code>
 */
public class EOByte extends EONode {
    public Byte b;

    public EOByte(Byte b) {
        this.b = b;
    }

    @Override
    public String generateEO(int indent) {
        return String.format("%02x", b).toUpperCase();
    }
}

File: src/main/java/eotree/data/EOByte.kt
package eotree.data

import eotree.EONode
import java.util.Locale

/**
 * EBNF representation:
 * `
 * /[\dA-F][\dA-F]/
` *
 */
class EOByte(var b: Byte) : EONode() {
    override fun generateEO(indent: Int): String {
        return String.format("%02x", b).uppercase(Locale.getDefault())
    }
}
